src/libaktualizr/crypto/keymanager.cc

```diff
@@ -189,7 +189,7 @@ std::string KeyManager::getCN() const {
   return cn;
 }
 
-void KeyManager::copyCertsToCurl(HttpInterface *http) {
+void KeyManager::copyCertsToCurl(const std::shared_ptr<HttpInterface> &http) {
```

Comment:
For this kind of function which doesn't share ownership and just use `http`, passing a simple pointer is ok. It reduces coupling with the rest of the code (doesn't depend on the flavor of memory management used by the caller).

See https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/ in section 3.:

> (a) and (b): Prefer passing parameters by * or &.

Reply:
If we hold an object or pointer, yes maybe raw pointer would be ok. But currently we already have a shared_ptr and I don't see any advantages here if I pass raw pointer from shared and pass it to the method. I have red this blog post, but for me it is not clear why raw pointer is better againts refference to the shared.

Reply:
In terms of generated code it should be the same, it's just that it's better if the type of pointer you pass serves as documentation for the way the object is used.

Passing a shared pointer indicates some ownership passing, otherwise you're not using it really. On the other hand, passing a raw pointer indicates that the method only makes use of the object for the duration of its execution (eg: borrow it).

But this is only a style matter, it should not block merging the PR.

Reply:
I see the point, but we are doing it wrong (according to that article) all over the place. For now I'd keep it consistent and we can debate if it's worth making the change to Sutter's recommendations. In the meantime, though, it looks like another rebase is unfortunately in order. And then I think we can merge.